src/nvim/buffer_defs.h

```diff
@@ -1058,10 +1073,16 @@ typedef struct {
   int zindex;
   WinStyle style;
   bool border;
+  bool title;
   bool shadow;
   schar_T border_chars[8];
   int border_hl_ids[8];
-  int border_attr[8];
+  int border_attr[9];
+  BorderTitlePos title_pos;
+  VirtText title_texts;
+  char *title_text;
+  int title_hi_id;
+  int title_attr;
```

Comment:
Set default values in `#define FLOAT_CONFIG_INIT` below.

`title_text` and `title_texts` can be a small problem. They use allocated memory that has to be freed later, probably in `win_free`. You'll also have to remember to free previous values when assigning new ones.

Reply:
yeah . there has some bug not fixed .just draft :)